File: backend/LegalService.API/Models/Entities/DocumentationService.cs
using System;
using System.Collections.Generic;

namespace LegalService.API.Models.Entities;

public class DocumentationService
{
    public int ServiceId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;

    // Soft-delete flag — deactivated services are hidden from customers
    public bool IsActive { get; set; } = true;

    // JSON list of required document labels, e.g. ["NIC","Address Proof","Business Reg Form"]
    // Stored as a plain string mapped to PostgreSQL TEXT; parsed by the application layer.
    public string RequiredDocuments { get; set; } = "[]";

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    // Navigation properties
    public ICollection<DocumentationRequest> DocumentationRequests { get; set; } = new List<DocumentationRequest>();
}

File: mobile/lib/models/documentation_service.dart
class DocumentationService {
  final int serviceId;
  final String name;
  final String description;
  final bool isActive;
  final List<String> requiredDocuments;

  DocumentationService({
    required this.serviceId,
    required this.name,
    required this.description,
    required this.isActive,
    required this.requiredDocuments,
  });

  factory DocumentationService.fromJson(Map<String, dynamic> json) {
    return DocumentationService(
      serviceId: json['serviceId'] is int ? json['serviceId'] : int.tryParse(json['serviceId'].toString()) ?? 0,
      name: json['name']?.toString() ?? '',
      description: json['description']?.toString() ?? '',
      isActive: json['isActive'] ?? true,
      requiredDocuments: (json['requiredDocuments'] as List<dynamic>?)
              ?.map((e) => e.toString())
              .toList() ??
          [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'serviceId': serviceId,
      'name': name,
      'description': description,
      'isActive': isActive,
      'requiredDocuments': requiredDocuments,
    };
  }
}
